lib/trino-parquet/src/main/java/io/trino/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -106,11 +110,31 @@ public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor
                 return createDecimalColumnReader(descriptor).orElse(new BinaryColumnReader(descriptor));
             case FIXED_LEN_BYTE_ARRAY:
                 return createDecimalColumnReader(descriptor)
-                        .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, " type FIXED_LEN_BYTE_ARRAY supported as DECIMAL; got " + descriptor.getPrimitiveType().getOriginalType()));
+                        .orElseGet(() -> {
+                            // Iceberg 0.11.1 writes UUID as FIXED_LEN_BYTE_ARRAY without logical type annotation (see https://github.com/apache/iceberg/pull/2913)
+                            if (isLogicalUuid(descriptor.getPrimitiveType()) || descriptor.getPrimitiveType().getLogicalTypeAnnotation() == null) {
```

Comment:
Breaking up this check into a method makes it hard to follow. We handle a null logical type both here and inside the method. Also, the method returns `false` if it's not annotated, then we turn around here and treat not annotated as a UUID.

If I'm understanding the code correctly, we could make the method return `true` for that case, so the entire logic is contained there.

Reply:
Is it expected that any primitive type other than `DECIMAL` is treated as an Iceberg `UUID`? What if it is `BOOLEAN`, `INT32`, etc.?

Reply:
FIXED_LEN_BYTE_ARRAY that is not recognized as decimal, is attempted to be read as uuid.
the reader later checks whether this was the correct decision.

The problem here is that we don't really know yet what will be the trino type we will be reading into.

Reply:
alternately, we could 
```
return createDecimalColumnReader(descriptor).orElse(new BinaryColumnReader(descriptor));
```

as the code for `BINARY` case is doing.  Would that be better?